### PR TITLE
Use the right query to list SharePoint sites and document libraries

### PR DESCRIPTION
**Pull request by zhedahht**
# Description

The query in my previous change (https://github.com/rubrikinc/polaris-o365-powershell/pull/16) can only list direct site children of a subscription, but can't list sites and document libraries recursively.

It uses the right query to list all sites and/or document libraries of a subscription, including the embedded ones.

## Related Issue

The query in my previous change (https://github.com/rubrikinc/polaris-o365-powershell/pull/16) can only list direct site children of a subscription, but can't list sites and document libraries recursively.

## How Has This Been Tested?

Manual tests

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/welcome-to-rubrik-build/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.

### CODE REVIEW

File: PolarisO365.ps1 [hunk 1]
```diff
@@ -820,170 +821,130 @@ function Get-PolarisO365SharePoint() {
...
+                    "field" = "NAME_OR_EMAIL_ADDRESS";
                     "texts" = @($SearchString);
                 }
-           )
+            );
             "first"     = 100;
-            "orgId"     = $SubscriptionId;
-            "sortBy"    = "NAME";
+            "o365OrgId" = $SubscriptionId;
+            "sortBy"    = "EMAIL_ADDRESS";
```
Comment (jlim1545) [inline PolarisO365.ps1:843]: EMAIL_ADDRESS does not apply to SharePoint object type, you should remove it.

File: PolarisO365.ps1 [hunk 2]
```diff
@@ -820,170 +821,130 @@ function Get-PolarisO365SharePoint() {
...
-                pageInfo {
-                    endCursor
-                    hasNextPage
-                    hasPreviousPage
-                }
-            }
+        $response = Invoke-RestMethod -Method POST -Uri $endpoint -Body $($payload | ConvertTo-JSON -Depth 100) -Headers $headers
+        $o365SharepointDrives = $response.data.o365SharepointDrives
+    } else {
+        $payload.query = "query o365SharepointDrives(`$after: String, `$o365OrgId:UUID!, `$filter: [Filter!], `$first: Int!, `$sortBy: HierarchySortByField, `$sortOrder: HierarchySortOrder) {
```
Comment (jlim1545) [inline PolarisO365.ps1]: Shouldn't we use the query for site not document libraries? The graphQL for elseif case and else case are same

File: QUICK_START.md [hunk 3]
```diff
@@ -80,6 +81,9 @@ $my_mailbox = Get-PolarisO365Mailbox -Token $token -PolarisURL $url -Subscriptio
 # get our onedrive user
 $my_onedrive = Get-PolarisO365OneDrive -Token $token -PolarisURL $url -SubscriptionId $my_sub.id -SearchString 'arif'
 
+# get our SharePoint sites
```
Comment (jlim1545) [inline QUICK_START.md:84]: Can we also add instruction for getting document library only or getting both items?

**Review by jlim1545 — CHANGES_REQUESTED** [review]

File: PolarisO365.ps1 (hunk 1 above)
Comment (zhedahht) [reply PolarisO365.ps1]: Removed variables related to sorting

File: PolarisO365.ps1 (hunk 2 above)
Comment (zhedahht) [reply PolarisO365.ps1]: The code was updated. Please review the latest version. Thanks.

**Review by jlim1545 — APPROVED** [review]
LGTM

**Pull request merged**